### PR TITLE
fix(ci): retry setup steps stopped by a failed tool download

### PR DESCRIPTION
**Pull request by JeffreyDallas**
## Description

Stacked on #5991 — base branch is `ci/retry-go-module-server-stalls`, so the diff here is only the two rules below. Merge #5991 first; GitHub will retarget this to `main`.

This pull request changes the following:

* Retries a setup step whose **release download was refused** with a 5xx or 429.
* Retries a setup step whose **download never reached its host**, as a resolver or socket error.
* Retries a setup step whose **download arrived corrupted** and failed its checksum.

A spell of 504s from GitHub's release downloads produced most of the unretried failures in a 48-hour window, in two shapes.

### The download is refused

```
##[error]Failed to download version v3.49.1: Error: Unexpected HTTP response: 504
Error: failed to fetch https://github.com/metallb/metallb/releases/download/metallb-chart-0.15.3/metallb-0.15.3.tgz : 504 Gateway Timeout
```

Eleven jobs, across six workflows:

| Failed step | Job | Workflow |
|---|---|---|
| `Install Task` | [Install Validation (debian)](https://github.com/hiero-ledger/solo/actions/runs/34856350585/job/104016802106) | Install Validation |
| `Install Task` | [Install Validation (podman)](https://github.com/hiero-ledger/solo/actions/runs/34857392068/job/104020368356) | Install Validation (Podman Runtime) |
| `Install Task` | [Standard Runner Deploy and Destroy](https://github.com/hiero-ledger/solo/actions/runs/34857392006/job/104020368650) | Standard Runner One-Shot |
| `Install Task` | [Windows Deploy and Destroy](https://github.com/hiero-ledger/solo/actions/runs/34856350540/job/104017085370) | Windows One-Shot |
| `Install Task` | [Docs Automation / CLI Help](https://github.com/hiero-ledger/solo/actions/runs/34857392467/job/104020370749) | PR Checks |
| `Install Task` | [Unit Tests / hiero-solo-linux-medium](https://github.com/hiero-ledger/solo/actions/runs/34857392467/job/104023325334) | PR Checks |
| `Install Task` | [Test Example (Version Upgrade Test)](https://github.com/hiero-ledger/solo/actions/runs/34857392053/job/104020542413) | Test Examples |
| `Setup E2E Tests` | [E2E Tests (Dual Cluster Full)](https://github.com/hiero-ledger/solo/actions/runs/34857392467/job/104023325387) | PR Checks |
| `Setup E2E Tests` | [E2E Tests (Dual Cluster Full)](https://github.com/hiero-ledger/solo/actions/runs/34856956142/job/104021845306) | PR Checks |

The `Setup E2E Tests` pair is worth noting: `step-security/retry` had already spent all three of its own attempts on the 504 before the step failed, so an in-step retry was not enough.

### The host cannot be reached at all

A download that never reaches its host looks different — the setup actions report a resolver error rather than an HTTP status. Both `Install Helm` jobs of run 34881026181 failed this way, each after the action had already retried once by itself:

```
Downloading 'v3.14.2' from 'https://get.helm.sh'
getaddrinfo EAI_AGAIN get.helm.sh
getaddrinfo EAI_AGAIN get.helm.sh
##[error]Error: Failed to download Helm from location https://get.helm.sh/helm-v3.14.2-linux-amd64.tar.gz
```

| Failed step | Job | Workflow |
|---|---|---|
| `Install Helm` | [Test Example (Multicluster Backup and Restore)](https://github.com/hiero-ledger/solo/actions/runs/34881026181/job/104170097799) | Test Examples |
| `Install Helm` | [Test Example (External Database Test)](https://github.com/hiero-ledger/solo/actions/runs/34881026181/job/104170097815) | Test Examples |

A **404 is deliberately not matched**, so a version that does not exist stays a settled failure — and the `Failed to download ... from location` wording alone is not enough, since that is what the action prints for a missing version too.

### The download is cut short

```
Installing kind...
sha256sum: WARNING: 1 computed checksum did NOT match
kind-linux-amd64: FAILED
```

kind is then never installed, and the action's own cleanup reports `kind: command not found`. Two jobs: [Install Kind](https://github.com/hiero-ledger/solo/actions/runs/34857392053/job/104020542460) (Test Examples) and [Setup Kind](https://github.com/hiero-ledger/solo/actions/runs/34856956142/job/104019227581) (PR Checks).

This is a truncated transfer, not the integrity claim a Go module mismatch makes — that one remains excluded by the `SECURITY ERROR` / `checksum mismatch` veto added in #5991.

| Rule | Step gate | Log must contain | Log must not contain |
|---|---|---|---|
| tool download could not reach its host | `install`, `setup`/`set up`, `configure`, `bootstrap`, `retry` | `Unexpected HTTP response: 5xx/429`, `failed to fetch …: 5xx`, `getaddrinfo EAI_AGAIN/ENOTFOUND`, `ECONNRESET`, `ETIMEDOUT` or `socket hang up` | `error TS####` |
| downloaded tool failed its checksum | same | `sha256sum: WARNING: N computed checksum(s) did NOT match` | `SECURITY ERROR` |

### Why both rules are gated to setup steps

The same 504 text appears in the logs of **15 jobs whose failing step runs the product rather than preparing it** — `Run E2E Tests`, `Run Example Test - *`, `Run GCS Test Script`. Those tests fetch charts too, so their logs carry the identical message while the failure itself may be real. The step gate refuses every one of them, so a test failure is never retried because a download misbehaved somewhere earlier in its log.

### Related Issues

* Stacked on #5991; related to #5954, #5967

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[x] This PR updated documentation
* \[x] This PR added no TODOs or commented out code
* \[x] This PR has no breaking changes
* \[x] Any technical debt has been documented as a separate issue and linked to this PR
* \[x] Any `package.json` changes have been explained to and approved by a repository manager
* \[x] All related issues have been linked to this PR
* \[x] All changes in this PR are included in the description
* \[x] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

The unchecked box: the repository has no harness for executing workflow YAML, so the logic was verified by simulating the embedded script against real GitHub API payloads and real job logs, as documented below.

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[x] These changes required manual testing that is documented below
* \[x] Anything not tested is documented

The following manual testing was done:

* **Whole-window sweep across three versions** — every watched failed run in a 48-hour window replayed with real job payloads and logs:

| Version | retried | refused |
|---|---|---|
| `main` (deployed) | 0 | 17 |
| #5991 | 0 | 17 |
| this PR | **4** | 13 |

  The four are [Install Validation (Podman Runtime)](https://github.com/hiero-ledger/solo/actions/runs/34857392068), [Standard Runner One-Shot](https://github.com/hiero-ledger/solo/actions/runs/34857392006), [Install Validation](https://github.com/hiero-ledger/solo/actions/runs/34856350585) and [Windows One-Shot](https://github.com/hiero-ledger/solo/actions/runs/34856350540) — all `Install Task` 504s.

* **Job-level match counts** — the download rule matches 11 jobs and the checksum rule 2, listed above. The remaining 13 runs stay refused because they mix a transient failure with a genuine one, which the existing mixed-failure guard requires.
* **Step-gate check** — 15 product-step jobs carry the same 504 text and are all refused, as intended.
* **The resolver signature** appears in 2 of the 61 failed job logs examined, both of them the `Install Helm` jobs above, and in no log of a step that runs the product. Replaying run [34881026181](https://github.com/hiero-ledger/solo/actions/runs/34881026181): `main` and the first commit of this PR both refuse it; with the broadened rule it is retried, reported as `tool download could not reach its host`. Verdicts across the rest of the window are unchanged (4 retried, 13 refused, before and after).
* **Static validation** — the YAML parses, the embedded `github-script` body parses as JavaScript, and the file is Prettier-clean.

The following was not tested:

* Live behaviour after merge — `workflow_run` only acts on the copy of the workflow on the default branch.
* Genuine failures in the same window are unaffected and still refused: `Check Code Style` prettier errors, `Launch Network with Prior Release 0.88.0`, `Run Example Test - Multicluster Backup and Restore`, and a Windows job that died with exit code `-1073741502` (`STATUS_DLL_INIT_FAILED`) — that last one is a runner-level fault seen once, and is left alone rather than guessed at.

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .github/workflows/zxf-transient-failure-retry.yaml [hunk 1]
```diff
@@ -92,6 +92,32 @@ jobs:
...
+                // Timeout"; the second surfaces as a resolver or socket error, as in `getaddrinfo
+                // EAI_AGAIN get.helm.sh`. `step-security/retry` and the setup actions have usually
+                // already spent their own attempts by then. A 404 is deliberately not matched,
+                // since a version that does not exist is a settled failure. Gated to steps that
+                // install or configure, so a product step that merely mentions one of these in its
+                // own output is left alone.
+                description: 'tool download could not reach its host',
+                stepPrefixes: ['install', 'setup', 'set up', 'configure', 'bootstrap', 'retry'],
+                required:
+                  /Unexpected HTTP response: (429|5\d\d)|failed to fetch [^\n]*: (429|5\d\d) |getaddrinfo (EAI_AGAIN|ENOTFOUND)|ECONNRESET|ETIMEDOUT|socket hang up/,
```
Comment (Fantomasa) [inline .github/workflows/zxf-transient-failure-retry.yaml]: **AI Suggestion:** 
question: the bare `ECONNRESET|ETIMEDOUT|socket hang up` tokens are tested against the whole job log, and `Setup E2E Tests` runs the 10 minute dual-cluster script, so a genuine failure there with a stray `ECONNRESET` earlier in the log would also be retried. Did the 61 logs show that shape anywhere? If not this is fine; otherwise anchoring them to the download line, like the artifact rule does, keeps the gate tight.

File: .github/workflows/zxf-transient-failure-retry.yaml [hunk 2]
```diff
@@ -92,6 +92,32 @@ jobs:
...
+                forbidden: /error TS\d{4}/,
+              },
+              {
+                // A download cut short arrives complete enough to write but fails its checksum, so
+                // the tool is never installed and the step's own cleanup then reports the binary as
+                // missing. Distinct from a Go module mismatch, which is an integrity claim about a
+                // published module and stays excluded above.
```
Comment (Fantomasa) [inline .github/workflows/zxf-transient-failure-retry.yaml]:
```suggestion
                // published module and stays excluded by the Go module rule below.
```
small one: the Go rule sits below these two now, so "above" points the wrong way.

File: .github/workflows/zxf-transient-failure-retry.yaml [hunk 3]
```diff
@@ -92,6 +92,32 @@ jobs:
...
+                description: 'downloaded tool failed its checksum',
+                stepPrefixes: ['install', 'setup', 'set up', 'configure', 'bootstrap', 'retry'],
```
Comment (Fantomasa) [inline .github/workflows/zxf-transient-failure-retry.yaml]: same list as line 106. Maybe hoist it into a `setupStepPrefixes` const above `rules`, so the two rules cannot drift apart?

File: .github/workflows/zxf-transient-failure-retry.yaml (hunk 1 above)
Comment (JeffreyDallas) [reply .github/workflows/zxf-transient-failure-retry.yaml]: Good catch — confirmed empirically: 0 of the 61 failed job logs from the audit window contain a bare `ECONNRESET`, `ETIMEDOUT`, or `socket hang up` anywhere, so the risk hadn't materialized here. Tightened it anyway in 67de5aff5 — all three are now anchored to the same line as a `fetch`/`download`/`install`/`request` word, matching how npm and http-client wrappers actually report a failed transfer (e.g. `npm ERR! network request to ... failed, reason: read ECONNRESET`). Re-verified against the real Helm run this rule targets and the full corpus: same 11 jobs matched, same 15 product-step jobs correctly refused.

File: .github/workflows/zxf-transient-failure-retry.yaml (hunk 3 above)
Comment (JeffreyDallas) [reply .github/workflows/zxf-transient-failure-retry.yaml]: Done in 67de5aff5 — hoisted into a single `setupStepPrefixes` const used by both rules.

**Review by jan-milenkov — APPROVED** [review]

**Review by Fantomasa — APPROVED** [review]
LGTM
